mvpa2/algorithms/group_clusterthr.py

```diff
@@ -0,0 +1,427 @@
+# emacs: -*- mode: python; py-indent-offset: 4; indent-tabs-mode: nil -*-
+# vi: set ft=python sts=4 ts=4 sw=4 et:
+### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+#
+#   See COPYING file distributed along with the PyMVPA package for the
+#   copyright and license terms.
+#
+### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+"""Cluster thresholding algorithm for a group-level searchlight analysis"""
+
+__docformat__ = 'restructuredtext'
+
+__all__ = ['GroupClusterThreshold',  'get_thresholding_map',
+           'get_cluster_sizes']
+
+if __debug__:
+    from mvpa2.base import debug
+
+import os
+import random
+import bisect
+from collections import Counter
+
+import numpy as np
+
+from scipy.ndimage import measurements
+from scipy.sparse import dok_matrix
+
+from mvpa2.mappers.base import IdentityMapper
+from mvpa2.datasets import Dataset
+from mvpa2.base.learner import Learner
+from mvpa2.base.param import Parameter
+from mvpa2.base.constraints import \
+        EnsureInt, EnsureFloat, EnsureRange, EnsureChoice
+from mvpa2.mappers.fx import mean_sample
+
+class GroupClusterThreshold(Learner):
+    """Statistical evaluation of group-level average accuracy maps
+
+    This algorithm can be used to perform cluster-thresholding of
+    searchlight-based group analyses. It implements a two-stage procedure that
+    uses the results of within-subject permutation analyses, estimates a per
+    feature cluster forming threshold (via bootstrap), and uses the thresholded
+    bootstrap samples to estimate the distribution of cluster sizes in
+    group-average accuracy maps under the NULL hypothesis, as described in [1]_.
+
+    Note: this class implements a modified version of that algorithm. The
+    present implementation differs in, at least, three aspects from the
+    description in that paper.
+
+    1) Cluster p-values refer to the probability of observing a particular
+       cluster size or a larger one (original paper: probability to observe a
+       larger cluster only).  Consequently, probabilities reported by this
+       implementation will have a tendency to be higher in comparison.
```

Comment:
wouldn't it be easy to provide option to perform as in original publication as well?  would ease comparison etc?